cygnus-common/src/main/java/com/telefonica/iot/cygnus/backends/ckan/CKANBackendImpl.java

```diff
@@ -681,4 +691,45 @@ private JsonResponse doCKANRequest(String method, String urlPath, String jsonStr
         return doRequest(method, urlPath, true, headers, new StringEntity(jsonString, "UTF-8"));
     } // doCKANRequest
 
+    public class DataStore{
+        private String resource_id;
+        private ArrayList<JsonElement> fields;
+        private String force;
+
+        public  DataStore(){}
+
+        public ArrayList<JsonElement> getFields() {
+            return fields;
+        }
+
+        public void setFields(ArrayList<JsonElement> fields) {
+            this.fields = fields;
+        }
+
+        public String getResource_id() {
+            return resource_id;
+        }
+
+        public void setResource_id(String resource_id) {
+            this.resource_id = resource_id;
+        }
+
+        public String getForce() {
+            return force;
+        }
+
+        public void setForce(String force) {
+            this.force = force;
+        }
+
+        @Override
+        public String toString() {
+            return "{" +
+                    "\"resource_id\":" + "\"" +resource_id + "\"" +
+                    ", \"fields\":" + "\""+fields+ "\""+
+                    ", \"force\":"+"\"" + force + "\"" +
+                    '}';
+        }
```

Comment:
Is this used? At any case, this is still a String Json. It should be optimized for gson.

Maybe you can use a JsonObject to build this String. Take a look to method NGSIUtils.linkedHashMapToJsonList.

Another solution would be creating a serializer and deserializer to indicate gson how to work with this object. take a look at NotifyContextRequestNGSIv2Deserializer.class and NotifyContextRequestNGSIv2.class. Both are used at NGSIRestHandler.class.

Reply:
That is only a toString method, it is no needed for the creation of the JSON, I am going to remove it.

Reply:
NTC